packages/wrangler/src/pages/functions/buildWorker.ts

```diff
@@ -362,7 +362,17 @@ const blockWorkerJsImports: Plugin = {
 					path: args.path,
 				};
 			}
-			// Otherwise, block any imports that the file is requesting
+			// If it's a node or cf built-in, mark it as external
+			if (
+				args.path.startsWith("node:") ||
```

Comment:
OK so I think we just need a bit of care here.
We should only consider imports beginning with `node:` as external if the `nodejs_compat` flag is actually turned on.

We can pass that through to this plugin by making it a function with a parameter, I believe.